### ISSUE TITLE
Unity bad code recognition when writing editor scripts - Assembly-CSharp-Editor

### ISSUE THREAD

**Issue by nice-shot** [OmniSharp/omnisharp-vscode#4247]
## Issue Description ##
When writing Unity editor scripts (scripts inside 'Editor' folders) using Visual Studio Code, omnisharp reports issues such as:

`The type or namespace name '_____' could not be found (are you missing a using directive or an assembly reference?) [Assembly-CSharp-Editor]`
and
`The name '_____' does not exist in the current context [Assembly-CSharp-Editor]`

It seems that unity generates a problematic Assembly-CSharp-Editor.csproj file.
Under `ProjectReference` you can see the line:
`<ReferenceOutputAssembly>false</ReferenceOutputAssembly>`

Commenting this line out for each ProjectReference fixes the issue. However, since unity generates this file, this issue returns when creating new scripts.

I'm not sure if this is Unity's responsibility or omnisharp but it would be very appreciated if this could be fixed.

## Steps to Reproduce ##

1. Configure VS Code as code editor for unity, based on the documentation here: https://code.visualstudio.com/docs/other/unity
2. Create new Unity project.
3. Create a new script "SampleBehaviour.cs".
4. Create an Editor folder.
5. Create a new script in the editor folder "SampleBehaviourEditor.cs".
6. Change its code to:

```
using UnityEditor;

[CustomEditor(typeof(SampleBehaviour))]
public class SampleBehaviourEditor : Editor
{}
```

7. Notice that omnisharp reports that 'SampleBehaviour' could not be found. Even though it will compile successfully in unity.

## Expected Behavior ##

Omnisharp should not report any issues and allow proper code completion for classes from the main assembly.

## Actual Behavior ##

Omnisharp reports issues about classes and objects from the main assembly and does not allow code completion and intellisense.

### OmniSharp log ###
<details>Post the output from Output-->OmniSharp log here</details>

### C# log ###
<details>Post the output from Output-->C# here</details>

## Environment information ##

**VSCode version**: 1.51.1
**C# Extension**: 1.23.6

<details><summary>Dotnet Information</summary>
.NET Core SDK (reflecting any global.json):
 Version:   3.1.404
 Commit:    470f6754b3

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.404\

Host (useful for support):
  Version: 3.1.10
  Commit:  1721e39439

.NET Core SDKs installed:
  2.1.403 [C:\Program Files\dotnet\sdk]
  3.1.404 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|csharp|ms-dotnettools|1.23.6|
|debugger-for-chrome|msjsdiag|4.12.11|
|docomment|k--kato|0.1.19|
|gitlens|eamodio|11.0.5|
|godot-tools|geequlim|1.1.2|
|jupyter|ms-toolsai|2020.11.372831992|
|python|ms-python|2020.11.371526539|
|spellright|ban|3.0.52|
|unity-debug|Unity|3.0.2|
|vscode-docker|ms-azuretools|1.8.1|
|vscode-standardjs|chenxsan|1.4.0|
|vsfire|toba|1.4.1|;

</details>

**Comment by filipw** [comment]
Make sure you updated VSCode Editor package in Unity to version 1.2.3 and regenerate all solution files. The earlier version generated incorrect project files with `<ReferenceOutputAssembly>false</ReferenceOutputAssembly>` in them.

The older OmniSharp didn't actually respect that setting so it was (accidentally) working fine. However, once we implemented correct handling for `ReferenceOutputAssembly` MSBuild property, that exposed the problem in VSCode Editor package in Unity, which was only fixed in version 1.2.3.

**Comment by nice-shot** [comment]
I updated the package and I see that this issue doesn't happen anymore. Thanks!
